### PR TITLE
Removes calls to isProduction outside of config

### PR DESCRIPTION
**Pull request by christopherjturner**

### CODE REVIEW

File: src/config/index.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ import { fileURLToPath } from 'node:url'
 
 const dirname = path.dirname(fileURLToPath(import.meta.url))
 
+const isProduction = process.env.NODE_ENV === 'production'
```
Comment (feedmypixel) [inline src/config/index.js:7]: Id also the others, `development` and `test` and update the config defaults for `isProduction`, `isDevelopment` and `isTest` to use these consts

File: src/helpers/logging/logger-options.js [hunk 2]
```diff
@@ -5,21 +5,19 @@ import { config } from '~/src/config/index.js'
...
   ignorePaths: ['/health'],
   redact: {
     paths: ['req.headers.authorization', 'req.headers.cookie', 'res.headers'],
     remove: true
   },
-  level: config.get('logLevel'),
-  ...(config.get('isDevelopment')
-    ? { transport: { target: 'pino-pretty' } }
-    : /** @type {Omit<LoggerOptions, 'mixin' | 'transport'>} */ (ecsFormat()))
+  level: config.get('log.level'),
```
Comment (feedmypixel) [inline src/helpers/logging/logger-options.js:15]: this is a bit bonkers. Can it be made clearer?

File: src/config/index.js (hunk 1 above)
Comment (feedmypixel) [reply src/config/index.js:7]:
```
const isProduction = process.env.NODE_ENV === 'production'
const isTest = process.env.NODE_ENV === 'test'
const isDevelopment = process.env.NODE_ENV === 'development'
```

Comment (feedmypixel) [reply src/config/index.js:7]:
```
  isProduction: {
    doc: 'If this application running in the production environment',
    format: Boolean,
    default: isProduction
  },
  isDevelopment: {
    doc: 'If this application running in the development environment',
    format: Boolean,
    default: isDevelopment
  },
  isTest: {
    doc: 'If this application running in the test environment',
    format: Boolean,
    default: isTest
  },
```

**Review by feedmypixel — COMMENTED** [review]
A few bits and bobs.

Some nice things in here that would be good to get into our `CDP` services

File: src/api/server.js [hunk 3]
```diff
@@ -40,16 +38,12 @@ async function createServer() {
...
-
-  if (isProduction) {
-    await server.register(secureContext)
-  }
-
-  // Pulse   - provides shutdown handlers
-  // mongoDb - sets up mongo connection pool and attaches to `server` and `request` objects
-  // router  - routes used in the app
-  await server.register([pulse, mongoDb, router])
+  // requestLogger - automatically logs incoming requests
```
[the remark this answers is not in the document]
Comment (christopherjturner) [reply src/api/server.js:41]: Looks like it all passes

File: src/helpers/logging/logger-options.js (hunk 2 above)
Comment (christopherjturner) [reply src/helpers/logging/logger-options.js:15]: possible solution?
```
/**
 * @type {{ecs: Omit<LoggerOptions, "mixin"|"transport">, "pino-pretty": {transport: {target: string}}}}
 */
const formatters = {
  ecs: ecsFormat(),
  'pino-pretty': { transport: { target: 'pino-pretty' } }
}

/**
 * @satisfies {Options}
 */
export const loggerOptions = {
  enabled: config.get('log.enabled'),
  ignorePaths: ['/health'],
  redact: {
    paths: ['req.headers.authorization', 'req.headers.cookie', 'res.headers'],
    remove: true
  },
  level: config.get('log.level'),
  ...formatters[config.get('log.format')]
}
```

Or could have two separate configs, one for ecs one for pino and just have the createLogger decide which to load?

Comment (feedmypixel) [reply src/helpers/logging/logger-options.js:15]: The above is a lot easier to read. I'd go with that. Lets see how that goes and then we have "Or could have two separate configs, one for ecs one for pino and just have the createLogger decide which to load?" as an option in the future?

**Pull request merged**